src/search/CMakeLists.txt

```diff
@@ -48,6 +56,18 @@ option(
 if(PLUGIN_LP_SOLVER_ENABLED AND USE_LP)
     find_package(OSI OPTIONAL_COMPONENTS Cpx Clp Grb Spx)
     if(OSI_FOUND AND (OSI_Cpx_FOUND OR OSI_Clp_FOUND OR OSI_Grb_FOUND OR OSI_Spx_FOUND))
+        if(USE_GLIBCXX_DEBUG)
+            message(
+                FATAL_ERROR
+                "Your build includes an LP solver and builds with the flag "
+                "GLIBCXX_DEBUG. This makes the planner binary incompatible "
+                "with the OSI library and leads to subtle errors. Set one of "
+                "the options USE_LP or USE_GLIBCXX_DEBUG to false. You can "
+                "also use the build configuration 'debuglp' for building with "
+                "LP solvers but without GLIBCXX_DEBUG or the configuration "
+                "'debugnolp' for building with GLIBCXX_DEBUG but without LP "
+                "solvers.")
```

Comment:
The message may be slightly misleading. _GLIBCXX_DEBUG *may* be compatible with OSI if OSI is also compiled with the flag. Also, this still uses the aliases we will remove, so I guess it needs some more rewording. Can you try some rewording, and then we have a look later? We may make this much shorter now and just say that the option USE_GLIBCXX_DEBUG is not supported when LP solvers are available and enabled. Perhaps point to the issue. I don't think we need a long explanation any more, now that this error can never be triggered by one of the standard aliases we include in build.py.

Reply:
I shortened the message to 
"The option USE_GLIBCXX_DEBUG is not supported when an LP solver is used to prevent binary incompatibilities (see issue982 for details)."

Reply:
Sounds a good improvement, but without further punctuations, I think this says that the LP solver is used to prevent binary incompabilities. Swap the order? "To prevent incompatibilities, the option USE_GLIBCXX_DEBUG is not supported when an LP solver is used. See issue982 for details."

Reply:
If we use an LP solver during our build process to prevent incompatibilities, the system is a bit over-engineered. :-)

I used the sentence you suggested.

Reply:
I am hazy on the details, but I remember reading that apt uses something like a SAT solver or CP solver or BDD library internally to manage certain dependencies. :-) (Not sure this is true; on quick search I found something related, like papers that describe apt alternatives using SAT solvers, but nothing conclusive regarding apt itself.)

Reply:
https://arxiv.org/pdf/1109.0113.pdf :-)